lint-python/action.yml

```diff
@@ -29,7 +29,7 @@ runs:
 
     - name: Run formatter
       if: steps.detect-ruff.outputs.ruff-available == 0
-      run: poetry run ruff format --force-exclude --check ${{ inputs.working_directory }}
+      run: poetry run ruff format --force-exclude --diff ${{ inputs.working_directory }}
```

Comment:
Sorry, a bit off-topic and I should have probably asked this in the pull request this was introduced, but why are we running both `ruff check` and then in a second step `ruff format`?

Reply:
> Sorry, a bit off-topic and I should have probably asked this in the pull request this was introduced, but why are we running both `ruff check` and then in a second step `ruff format`?

Maybe I didn't understand you question, but here the explanation that I can give: `ruff check` is a linter (like flake8 with plugins), whereas the `ruff format` formats the code similar to `black`. We could run them together like `ruff check && ruff format`, but don't do it to not mix outputs.

Reply:
Thanks, this what I am asking, but it seems like I'm missing something still. Is there a situation where `ruff check` passes, but `run format --check` returns a non-zero exit code? I don't think we ever ran `black` in pipelines, so I'm wondering why a step to run formatting (as opposed to just checking) is necessary now.

Reply:
Here is an example:
```python
a={"a":1}
```
`ruff check` returns zero exit code, while `ruff format` returns an error because it's not properly formatted (missing spaces around the equal sign and after the colon).

Reply:
Thank you, @catcombo! Flake8 also did this kind of checking (`"black would make changes..."`), so I was surprised that ruff separates this in different steps. Thank you for the example, this helped a lot.

Reply:
> Flake8 also did this kind of checking (`"black would make changes..."`), so I was surprised that ruff separates this in different commands.

`flake8` doesn't provide this check by default, so we used https://github.com/peterjc/flake8-black plugin. This made it possible to run all checks in one command.

Reply:
Makes sense! What is the conclusion for this pull request, then? I guess we can just wait for @zyv to be back from his holiday. In general, we can add `--diff` to `ruff check`, but I'm not sure if there is a way to get diffs for `ruff format`.

Reply:
> `--diff` flag doesn't exist for the ruff format command. You have confused it with the ruff check command.

My previous comment was wrong because I used an old `ruff` version. It seems `--diff` flag was introduced in the latest releases. I tested it locally and it works as expected. I don't know why Yury want to see diff in CI logs. For me standard output is enough.

```bash
> ruff format --force-exclude --check .
Would reformat: demo.py
```

```bash
> ruff format --force-exclude --diff .
--- demo.py
+++ demo.py
@@ -1 +1 @@
-a={"a":1}
\ No newline at end of file
+a = {"a": 1}

--- moneymeets_backoffice/widgets.py
+++ moneymeets_backoffice/widgets.py
@@ -65,6 +65,5 @@
         attrs=None,
     ):
         return super().create_option(name, value, label, selected, index, subindex, attrs) | {
-            "label":
-                mark_safe(   f'<a href="{self.attrs["urls"][value]}" target="_blank">{label}</a>'),
+            "label": mark_safe(f'<a href="{self.attrs["urls"][value]}" target="_blank">{label}</a>'),
         }
```

I personally don't need this. Let's wait until Yury is back.

Reply:
I want to see the diff for the same reason why `check-git-diff` is called after formatting JSON:

https://github.com/moneymeets/moneymeets-composite-actions/blob/master/lint-python/action.yml#L19

... and why it is called everywhere, where we generate code from the specs:

https://github.com/search?q=org%3Amoneymeets%20check-git-diff&type=code

If I get an email message that the build has failed, and I go as far as to click on the link to see the logs to find out that it's a formatting issue, it's helpful to me to see directly what exactly is the problem, especially since GitHub colors it in the nice way.

Reply:
Thanks for explanation. If it's ok for everyone else, you can merge it.